### PR TITLE
refactor(source): refactor avro/protobuf parser and support enum type by mapping to var…

### PR DESCRIPTION
**Pull request by waruto210**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

- refactor avro/protobuf parser to make their logic clearer
- support enum type in avro/protobuf parser

## Checklist

~- [] I have written necessary rustdoc comments~
~- [ ] I have added necessary unit tests and integration tests~
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

Now user can define enum fields in the avro/protobuf schema 

### Types of user-facing changes

* Connector (sources & sinks)

## Refer to a related PR or issue link (optional)
#5840

### CODE REVIEW

File: src/source/src/parser/pb_parser.rs [hunk 1]
```diff
@@ -134,102 +134,70 @@ impl ProtobufParser {
...
-        }
-        DataType::Float64 => {
-            protobuf_match_type!(value, ScalarImpl::Float64, { I32, U32, F32, F64}, OrderedF64)
-        }
-        DataType::Decimal => {
-            protobuf_match_type!(value, ScalarImpl::Decimal, { I32, I64, U32, U64}, Decimal)
+        Value::Message(dyn_msg) => {
+            let mut rw_values = Vec::with_capacity(dyn_msg.descriptor().fields().len());
+            // fields is a btree map in descriptor
+            // so it's order is the same as datatype
```
Comment (tabVersion) [inline src/source/src/parser/pb_parser.rs:165]: this finding can save some work, well done.

File: src/source/src/parser/pb_parser.rs [hunk 2]
```diff
@@ -134,102 +134,70 @@ impl ProtobufParser {
...
+            ScalarImpl::Struct(StructValue::new(rw_values))
         }
-        DataType::Varchar => {
-            protobuf_match_type!(value, ScalarImpl::Utf8, { String }, String)
+        Value::List(values) => {
+            let rw_values = values
+                .iter()
+                .map(|value| from_protobuf_value(field_desc, value))
+                .collect::<Result<Vec<_>>>()?;
+            tracing::info!("list {}, value {:?}", field_desc.name(), rw_values);
```
Comment (tabVersion) [inline src/source/src/parser/pb_parser.rs:189]: consider change to debug here

**Review by tabVersion — APPROVED** [review]
rest LGTM

File: src/source/src/parser/pb_parser.rs (hunk 2 above)
Comment (waruto210) [reply src/source/src/parser/pb_parser.rs:189]: LOL. Actually, I forgot to delete this

**Pull request merged**